packages/@aws-cdk/aws-stepfunctions-tasks/lib/eventbridge/put-events.ts

```diff
@@ -0,0 +1,165 @@
+   * i.e. "{ \"instance-id\": \" i-1234567890abcdef0\", \"state\": \"terminated\" }"
+   * @default - service dependent
+   */
+  readonly detail?: sfn.TaskInput;
+
+  /**
+   * Along with the source field help identify fields and values of detail field
+   * For example, events by CloudTrail have detail type "AWS API Call via CloudTrail"
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   * @default - service dependent
+   */
+  readonly detailType?: string;
+
+  /**
+   * The event bus the entry will be sent to.
+   *
+   * @default - event is sent to account's default event bus
+   */
+  readonly eventBus?: events.IEventBus;
+
+  /**
+   * JSON array containing ARNs of resources used in the event
+   *
+   * For example, auto scaling events include ARNs for both instances and auto scaling groups
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   * @default []
+   */
+  readonly resources?: string[];
+
+  /**
+   * The service that generated the event such as com.mycompany.myapp
+   * @example 'com.example.service'
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   * @default - service dependent
+   */
+  readonly source?: string;
+
+  /**
+   * Timestamp of the event entry
+   * @default Timestamp of the PutEvents call
+   */
+  readonly timestamp?: number;
+}
+
+/**
+ * Properties for sending events with PutEvents
+ */
+export interface EventBridgePutEventsProps extends sfn.TaskStateBaseProps {
+  /**
+   * The entries that will be sent (must be at least 1)
+   */
+  readonly entries: EventBridgePutEventsEntry[];
+}
+
+/**
+ * A StepFunctions Task to send events to an EventBridge event bus
+ */
+export class EventBridgePutEvents extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EventBridgePutEventsProps) {
+    super(scope, id, props);
+
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EventBridgePutEvents.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN) {
+      if (!sfn.FieldUtils.containsTaskToken(props.entries)) {
+        throw new Error('Task Token is required in `entries`. Use JsonPath.taskToken to set the token.');
+      }
+    }
+
+    this.validateEntries();
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        actions: ['events:PutEvents'],
+        resources: this.eventBusArns,
```

Comment:
do we run into a deployment error if this array is empty (only default event bus in entries)?

Reply:
Reran integration test with no event bus passed and still deployed and ran successfully, it grabbed the default bus as per line 109.